### PR TITLE
GH-655 Add catboy command. Include afk event state to flow. Add Teleport API

### PR DESCRIPTION
**Pull request by Rollczi**
image

image

https://cdn.discordapp.com/attachments/873137700448911380/1203468848804794378/20240203-2235-08.8582817.mp4?ex=65d134b4&is=65bebfb4&hm=7f43967b188ded5f682bf1c7a4c2279693cd3f3c32b8916f52dbcb09fadb8532&

### CODE REVIEW

**Review by igoyek — APPROVED** [review]
Najlepszy ficzer tej wtyczki 👌😉

File: eternalcore-api/src/main/java/com/eternalcode/core/EternalCoreApi.java [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 import com.eternalcode.core.feature.spawn.SpawnService;
+import com.eternalcode.core.feature.teleport.TeleportService;
 
 public interface EternalCoreApi {
 
     AfkService getAfkService();
 
     SpawnService getSpawnService();
 
+    CatboyService getCatboyService();
```
Comment (P3ridot) [inline eternalcore-api/src/main/java/com/eternalcode/core/EternalCoreApi.java:14]: Best service I've ever seen

File: eternalcore-api/src/main/java/com/eternalcode/core/feature/catboy/Catboy.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.eternalcode.core.feature.catboy;
+
+import org.bukkit.entity.Cat;
+
+import java.util.UUID;
+
+public record Catboy(UUID uuid, Cat.Type selectedType) {
```
Comment (P3ridot) [inline eternalcore-api/src/main/java/com/eternalcode/core/feature/catboy/Catboy.java:7]: Fabulous utilization of modern Java API

File: eternalcore-api/src/main/java/com/eternalcode/core/feature/catboy/event/CatboyChangeTypeEvent.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+
+/**
+ * Called when a player switches their catboy type.
+ */
+public class CatboyChangeTypeEvent extends PlayerEvent implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
```
Comment (P3ridot) [inline eternalcore-api/src/main/java/com/eternalcode/core/feature/catboy/event/CatboyChangeTypeEvent.java:14]: Rename to CATBOY_HANDLER_List for better catboy feature naming integrity

File: eternalcore-api/src/main/java/com/eternalcode/core/feature/catboy/event/CatboySwitchEvent.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+
+/**
+ * Called when a player switches their catboy status.
+ */
+public class CatboySwitchEvent extends PlayerEvent {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
```
Comment (P3ridot) [inline eternalcore-api/src/main/java/com/eternalcode/core/feature/catboy/event/CatboySwitchEvent.java:12]: Same like in previous event

**Comment by Rollczi** [comment]
Real review
image

**Review by P3ridot — APPROVED** [review]
Ready to merge, however next time don't mix so life changing changes like catboys with so unimportant ones like some teleport api

File: eternalcore-api/src/main/java/com/eternalcode/core/EternalCoreApi.java (hunk 1 above)
Comment (Rollczi) [reply eternalcore-api/src/main/java/com/eternalcode/core/EternalCoreApi.java:14]: 😎

**Comment by Rollczi** [comment]
> Ready to merge, however next time don't mix so life changing changes like catboys with so unimportant ones like some teleport api

Yes sir!

**Review by CitralFlo — APPROVED** [review]
/| _ ╱|、  
             ( •̀ㅅ •́  )
       ＿ノ ヽ ノ＼＿ 
    /　`/ ⌒Ｙ⌒ Ｙ　 \
 ( 　(三ヽ人　 /　 　|
|　ﾉ⌒＼ ￣￣ヽ　 ノ
ヽ＿＿＿＞､＿＿／
          ｜( 王 ﾉ〈 
           /ﾐ`ー―彡\ 
          |╰          ╯|   
          |       /\       |
          |      /  \      |                    
          |    /     \     |

**Review by CitralFlo — APPROVED** [review]

**Comment by Jakubk15** [comment]
Liberum veto

**Review by Jakubk15 — APPROVED** [review]
nie no żartuję, dobry ficzer Rollek

**Review by vLuckyyy — APPROVED** [review]

**Review by imDMK — APPROVED** [review]
🐱

**Pull request merged**